appserver/packager/glassfish-jpa/pom.xml

```diff
@@ -95,6 +95,11 @@
            <artifactId>persistence-common</artifactId>
            <version>${project.version}</version>
         </dependency>
+        <dependency>
+            <groupId>fish.payara.server.internal.persistence</groupId>
+            <artifactId>data-core</artifactId>
+            <version>${project.version}</version>
+        </dependency>
```

Comment:
See my other comment, I wouldn't necessarily have this as a part of persistence as it's not a part of JPA (just interacts with it).

I would have it as a standalone `data` package